crates/service/src/tap/receipt.rs

```diff
@@ -0,0 +1,74 @@
+// Copyright 2023-, Edge & Node, GraphOps, and Semiotic Labs.
+// SPDX-License-Identifier: Apache-2.0
+
+use tap_core::{
+    receipt::{WithUniqueId, WithValueAndTimestamp},
+    signed_message::SignatureBytes,
+};
+use thegraph_core::alloy::{dyn_abi::Eip712Domain, primitives::Address, signers::Signature};
+
+#[derive(Debug, Clone, PartialEq, Eq)]
+pub enum TapReceipt {
+    V1(tap_graph::SignedReceipt),
+    V2(tap_graph::v2::SignedReceipt),
+}
```

Comment:
For my edification, why would this not be in, say `tap_core` or some common library. I'm not saying tap-core is the right place just wondering about the common libs for tap 🙏